components/bl-credit-card-component/src/helpers.js

```diff
@@ -0,0 +1,313 @@
+const eloStringPattern = '^401178|^401179|^431274|^438935|^451416|^457393|^457631|^457632|^504175|^627780|^636297' +
+  '|^636369|^636368|^(506699|5067[0-6]\\d|50677[0-8])|^(50900\\d|5090[1-9]\\d|509[1-9]\\d{2})|^65003[1-3]' +
+  '|^(65003[5-9]|65004\\d|65005[0-1])|^(65040[5-9]|6504[1-3]\\d)|^(65048[5-9]|65049\\d|6505[0-2]\\d|65053[0-8])' +
+  '|^(65054[1-9]|6505[5-8]\\d|65059[0-8])|^(65070\\d|65071[0-8])|^65072[0-7]|^(65090[1-9]|65091\\d|650920)' +
+  '|^(65165[2-9]|6516[6-7]\\d)|^(65500\\d|65501\\d)|^(65502[1-9]|6550[3-4]\\d|65505[0-8])|^(65092[1-9]|65097[0-8])';
+
+const eloPattern = new RegExp(eloStringPattern);
+const defaultNumberFormat = /(\d{1,4})(\d{1,4})?(\d{1,4})?(\d{1,4})?/;
+const longNumberFormat = /(\d{1,4})(\d{1,4})?(\d{1,4})?(\d{1,7})?/;
+
+const Cards = [
+  {
+    type     : 'amex',
+    pattern  : /^3[47]/,
+    format   : /(\d{1,4})(\d{1,6})?(\d{1,5})?/,
+    length   : [15],
```

Comment:
All of that taken from well quality library with tests coverage?

Reply:
I got this from the [Payment ](https://github.com/jessepollak/payment) library. This library is used in others that have more than 10k stars.